typpete/src/annotation_resolver.py

```diff
@@ -286,15 +291,19 @@ def unparse_annotation(self, z3_type):
                     instance = simplify(accessor(z3_type))
                     args.append(self.unparse_annotation(instance))
                 return '{}[{}]'.format(generic_name, ', '.join(args))
-            return match.group(1)
+            annotation = match.group(1)
+            if annotation in definition_linenos and (lineno <= definition_linenos[annotation]
+                                                     or context_name == annotation):
+                annotation = "'" + annotation + "'"
```

Comment:
So if I read this correctly, we won't string-escape type annotations for types which haven't been imported at all, right? Intuitively I'd think we should do that.

Reply:
Well we do import every type that we provide as an annotation

Reply:
Ah okay, I didn't know that.